packages/theme/src/cli/services/update/run.ts

```diff
@@ -0,0 +1,178 @@
+            url: themePreviewUrl(ctx.theme, session),
+          },
+        },
+      ],
+    ],
+  })
+}
+
+async function check(ctx: Context, options: Options) {
+  const {script} = options
+
+  await sleep(1)
+  await checkScript(script)
+
+  ctx.scriptContent = await readFile(joinPath(script))
+}
+
+async function triggerUpdater(ctx: Context, session: AdminSession, options: Options) {
+  const {'source-theme': source, 'target-theme': target} = options
+  const store = session.storeFqdn
+
+  const themes = await fetchStoreThemes(session)
+
+  const sourceThemeId = findThemeByIdentifier(store, themes, source).id
+  const targetThemeId = findThemeByIdentifier(store, themes, target).id
+
+  // This is a false positive, as `renderTasks` performs tasks sequentially.
+  // eslint-disable-next-line require-atomic-updates
+  ctx.theme = await triggerUpdaterAPI(session, ctx.scriptContent, sourceThemeId, targetThemeId)
+}
+
+async function waitForUpdater(ctx: Context, session: AdminSession) {
+  const startTime = Date.now()
+
+  /**
+   * Generally, it's a good practice to take full advantage of the parallelization
+   * benefits of async/await.
+   *
+   * However, in the context of polling an API, each iteration must wait for the
+   * response, before the next polling.
+   */
+  // eslint-disable-next-line no-await-in-loop
+  while (await isUpdaterIsProgress(ctx, session, startTime)) {
+    // retry
+  }
+}
+
+async function isUpdaterIsProgress(ctx: Context, session: AdminSession, startTime: number) {
+  const theme = await fetchTheme(ctx.theme.id, session)
+
+  if (!theme) {
+    throw new AbortError(
+      'The `update_extension.json` script could not be executed due to a runtime issue.',
+      'Please check the `update_extension.json` script and retry.',
+    )
+  }
+
+  const elapsedTime = Date.now() - startTime
+
+  if (theme.processing && elapsedTime >= UPDATER_TIMEOUT) {
+    throw new AbortError(
+      'The `update_extension.json` script could not be executed due to a timeout issue.',
+      'Please check the `update_extension.json` script and retry.',
+    )
+  }
+
+  // Sleep for 1 second before polling again
+  await sleep(1)
+
+  return theme.processing
+}
+
+/**
+ * FIXME: Trigger the Updater API instead of creating a theme
+ *
+ * Trigger the Updater API
+ *
+ * @param session - current admin session.
+ * @param _updateExtension - `update_extension.json` script content.
+ * @param source - The theme ID or name of the theme at the previous version.
+ * @param target - The theme ID or name of the theme at the target version.
+ *
+ * @returns the reference to the updated theme.
+ */
+async function triggerUpdaterAPI(
+  session: AdminSession,
+  _updateExtension: string,
+  source: number,
+  target: number,
+): Promise<Theme> {
+  const name = `updated-theme-${source}-${target}`
+  const role = 'unpublished'
+  try {
+    const theme = await createTheme({name, role, processing: true}, session)
```

Comment:
Will this be a future change?

Reply:
Yes. It may seem strange to call `createTheme` within the `updateTheme` flow, but we do so because we're actually creating a new-updated theme (we're not modifying an existing resource).

At the moment, this PR is not doing that as it depends on a new API. However, the plan is having that end-to-end before merging the changes.